test/integration/suites/upgrade/00-setup

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -e
+
+mkdir -p shared/server-data
+mkdir -p shared/agent-data
+mkdir -p test/before-server-upgrade
+mkdir -p test/after-server-upgrade
+mkdir -p test/after-agent-upgrade
```

Comment:
The suite directory is copied in its entirety to a temporary directory where the actual test is run so that the tests can (ab)use the current working directory as needed. The integration test framework does cleanup of that directory after executing 99-teardown.